### PR TITLE
feat: blueprint-test file logger

### PR DESCRIPTION
**Pull request by bharathkkb**
This adds a new logger that can log to disk instead of stdout. This helps keep cloudbuild logs tidier with additional details in disk and uploaded using [CB artifacts](https://cloud.google.com/build/docs/building/store-artifacts-in-cloud-storage#store-artifact-storage).

### CODE REVIEW

**Comment by bharathkkb** [comment]
cc @daniel-cit

**Comment by daniel-cit** [comment]
@bharathkkb thanks for the PR

one question, is this new logger to be used with

https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/4e9b833071f920c6dd3fbb00771522dd4782c543/infra/blueprint-test/pkg/tft/terraform.go#L147

in this case we have code that does not use the custom logger:

https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/1b6b73f036bfb3e864141fa14d762354d71b9ab9/infra/blueprint-test/pkg/utils/poll.go#L41

https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/1b6b73f036bfb3e864141fa14d762354d71b9ab9/infra/blueprint-test/pkg/utils/poll.go#L43

do they need to be addressed here or in a new PR?

File: infra/blueprint-test/pkg/utils/logger_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			content: "foo",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+			testFile := path.Join(t.TempDir(), fmt.Sprintf("%s.log", tt.name))
+			fl, flClose := NewTestFileLogger(t, testFile)
+			fl.Logf(t, tt.content)
+			flClose(t)
```
Comment (daniel-cit) [inline infra/blueprint-test/pkg/utils/logger_test.go:44]: would actual usage be something like:

```go
fl, flClose := NewTestFileLogger(t, testFile)
bpt := tft.NewTFBlueprintTest(t,
	tft.WithLogger(fl),
)
defer flClose(t)
```

**Comment by bharathkkb** [comment]
> in this case we have code that does not use the custom logger:

This is primarily for TF logging, utils wont be able to use this logger yet. We can re evaluate if this is needed.

File: infra/blueprint-test/pkg/utils/logger_test.go (hunk 1 above)
Comment (bharathkkb) [reply infra/blueprint-test/pkg/utils/logger_test.go:44]: Yup that looks right.

**Review by g-awmalik — APPROVED** [review]

**Pull request merged**